vmm/src/memory_manager.rs

```diff
@@ -450,10 +503,32 @@ impl MemoryManager {
         Ok(slot)
     }
 
+    pub fn virtiomem_resize(&mut self, size: u64) -> Result<(), Error> {
+        let region = self.virtiomem_region.take();
+        if let Some(region) = region {
+            self.add_region(region)?;
+        }
+
+        if let Some(resize) = &self.virtiomem_resize {
+            resize.work(size).map_err(Error::VirtioMemResizeFail)?;
+        } else {
+            panic!("should not fail here");
+        }
+
+        Ok(())
+    }
+
     pub fn resize(&mut self, desired_ram: u64) -> Result<bool, Error> {
-        if desired_ram > self.current_ram {
-            self.hotplug_ram_region((desired_ram - self.current_ram) as usize)?;
-            self.current_ram = desired_ram;
+        if desired_ram >= self.current_ram {
+            let size = desired_ram - self.current_ram;
+            if self.virtiomem_resize.is_some() {
+                self.virtiomem_resize(size)?;
```

Comment:
@teawater here you should `return Ok(false)` after the `virtiomem_resize()` succeeded. As @sameo pointed out to me, in case of `virtio-mem`, we don't want the ACPI hotplug notification to be triggered as it would be done [here](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/master/vmm/src/vm.rs#L550-L562).
So by simply returning `false`, we're indicating that we don't want the ACPI notification to be triggered.

Reply:
Yes, that would help making sure we don't send the GED notification to the guest when using virtio-mem.
I think a cleaner solution would be, from `vm.resize()`, to call the GED notification only when the hotplug method is ACPI. Something like that:

```Rust
            if self
                .memory_manager
                .lock()
                .unwrap()
                .resize(desired_memory)
                .map_err(Error::MemoryManager)?
            {
                let memory_config = self.config.lock().unwrap().memory.clone();
                match memory_config.hotplug_method {
                    HotplugMethod::Acpi => {
                        self.device_manager
                            .lock()
                            .unwrap()
                            .notify_hotplug(HotPlugNotificationFlags::MEMORY_DEVICES_CHANGED)
                            .map_err(Error::DeviceManager)?;
                    }
                    HotplugMethod::VirtioMem => {}
                }
            }
            self.config.lock().unwrap().memory.size = desired_memory;
```

Reply:
This is a good idea.

Reply:
It is fixed in the last version.